middlewares/ownership.js

```diff
@@ -0,0 +1,21 @@
+const createHttpError = require('http-errors')
+const { catchAsync } = require('../helpers/catchAsync');
+const { ErrorObject } = require("../helpers/error");
+
+module.exports = {
+  ownership: catchAsync(async (req, res, next) => {
+    try {
+
+      if (req.user.id !== parseInt(req.params.id) && req.user.roleId !== 1)
```

Comment:
Make sure that you are in the scope of card because when the user id is the same that the user id params it should be go and in this case it need be admin in all case

Reply:
I understand that when the user id is the same that the user id params it should be go, it doesn't matter is "admin" or "user". When they are different, in this case it need be "admin".